### DISCUSSION TITLE
Functionality of secrets

### DISCUSSION THREAD

**Discussion by gblyshchyk** [netbox-community/netbox#5886] [Ideas]
There are a few notes(or questions) on using secrets:
1. Is it possible to make sure that the secret is not shown on the screen when you press the Unlock button? For example, when people are nearby or you record actions with a screen capture program, this is bad.
2. The password is transmitted in clear text via an API request. Is it possible to make it so that it would be sent encrypted and decrypted on the client side?

**Comment by jeremystretch** [comment]
1. Not currently, no. Unfortunately the secrets functionality in NetBox will likely be removed in the near future, so not much development is being done with it at the moment.

2. This is addressed by ensuring the UI and API are accessible only using HTTPS.

**Reply by gblyshchyk** [reply]
Thank you for answer

**Reply by proudbro** [reply]
@jeremystretch , Hi, Jeremy. Will the secrets functionality be completely cleaned out from Netbox or replaced with something else?

**Reply by jeremystretch** [reply]
Please see #5278
